plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanelTest.java

```diff
@@ -163,28 +169,35 @@ public void testUncheckStopPreviousVersionButtonWhenDisabled() {
   }
 
   @Test
-  public void testProjectSavedInPreferencesSelected() throws ProjectRepositoryException {
+  public void testProjectSavedInPreferencesSelected()
+      throws ProjectRepositoryException, InterruptedException, BackingStoreException {
     IEclipsePreferences node =
         new ProjectScope(project).getNode(StandardDeployPreferences.PREFERENCE_STORE_QUALIFIER);
     node.put("project.id", "projectId1");
     node.put("account.email", EMAIL_1);
     initializeProjectRepository();
     when(loginService.getAccounts()).thenReturn(new HashSet<>(Arrays.asList(account1, account2)));
     deployPanel = createPanel(true /* requireValues */);
+    deployPanel.latestGcpProjectQueryJob.join();
+    node.clear();
+
+    ProjectSelector projectSelector = getProjectSelector();
+    IStructuredSelection selection = projectSelector.getViewer().getStructuredSelection();
+    assertThat(selection.size(), is(1));
+    assertThat(((GcpProject) selection.getFirstElement()).getId(), is("projectId1"));
+  }
+
+  private ProjectSelector getProjectSelector() {
     Queue<Control> children = new ArrayDeque<>(Arrays.asList(deployPanel.getChildren()));
     while (!children.isEmpty()) {
       Control control = children.poll();
       if (control instanceof ProjectSelector) {
-        ProjectSelector projectSelector = (ProjectSelector) control;
-        IStructuredSelection selection = projectSelector.getViewer().getStructuredSelection();
-        assertThat(selection.size(), is(1));
-        assertThat(((GcpProject) selection.getFirstElement()).getId(), is("projectId1"));
-        return;
+        return (ProjectSelector) control;
       } else if (control instanceof Composite) {
         children.addAll(Arrays.asList(((Composite) control).getChildren()));
       }
-    };
-    fail("Did not find ProjectSelector widget");
+    }
+    return null;
```

Comment:
I'd keep the `fail()` here as there's not such test case where `null` is an expected outcome.